sftp/client.go

```diff
@@ -84,7 +84,7 @@ type clientImpl struct {
 }
 
 type Client interface {
-	Create(path string) (io.WriteCloser, error)
+	OpenFile(path string, f int) (io.WriteCloser, error)
 	Open(path string) (io.ReadCloser, error)
```

Comment:
Instead of having two methods, `Open` for `reading` and `OpenFile` for `writing`, we can simply have one method with the flag parameter which will determine where we opened the files for reading or writing.
```suggestion
	Open(path string, flag int) (io.ReadWriteCloser, error)
```

Reply:
During download, we would use something like below:
```GO
// Download downloads a file from the remote server
func (fm *fileManagerImpl) Download(remoteFilePath, localDir string) error {
	remoteFile, err := fm.client.Open(remoteFilePath, os.O_RDONLY)
	if err != nil {
		return fmt.Errorf("cannot open remote file: %w", err)
	}
	defer func() {
		_ = remoteFile.Close()
	}()
```

Reply:
Also, in tests we would need to modify nopWriteCloser to something like this, since we don't have anything like `io.NopReadWriteCloser`
```GO
type nopReadWriteCloser struct {
	io.ReadWriter
}

func (nwc *nopReadWriteCloser) Close() error {
	return nil
}
```

Reply:
Something like https://github.com/rudderlabs/rudder-go-kit/pull/456

Reply:
addressed